### DISCUSSION TITLE
VSCode For Web Can't Open Folder - but add works

### DISCUSSION THREAD

**Discussion by jlewi** [microsoft/vscode-discussions#990] [Extension Development QnA]
Hi 

I'm self-hosting vscode for web. Opening a folder doesn't work; but adding does.

The chrome console shows the following error

```
ERR Can't open folders, try adding a folder to the workspace instead.: Error: Can't open folders, try adding a folder to the workspace instead.
    at a.pickFolderAndOpen (https://mydomain/out/vs/workbench/workbench.web.main.js:1891:5002)
```

Does anyone have any pointers on what could be causing this? Or how to debug it?

Thank you

**Comment by jlewi** [comment]
If I add a query argument like `?workspace=vscode-userdata%3A%2FSome%2Fsubdir%2Fwfname` then it appears to work. The query argument seems to be defining a workspace with the name given by the final element of the path encoded in the argument.

This seems vaguely like this old issue microsoft/vscode#124494 with Open Folder not working in an empty workspace.

**Comment by New-templates1990** [comment]
> Hi
> 
> I'm self-hosting vscode for web. Opening a folder doesn't work; but adding does.
> 
> The chrome console shows the following error
> 
> ```
> ERR Can't open folders, try adding a folder to the workspace instead.: Error: Can't open folders, try adding a folder to the workspace instead.
>     at a.pickFolderAndOpen (https://mydomain/out/vs/workbench/workbench.web.main.js:1891:5002)
> ```
> 
> Does anyone have any pointers on what could be causing this? Or how to debug it?
> 
> Thank you

**Comment by olguzzar** [comment]
Hi @jlewi, thanks for your question. This forum is specifically for extension developers. I'd recommend you post this in the [VS Code repo](https://github.com/microsoft/vscode/issues) to get more relevant eyes on it.

**Comment by progrium** [comment]
Was this issue added to the VS Code repo? I'm having trouble finding it.
